packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -149,6 +149,18 @@ export interface FunctionAttributes {
    * to this Lambda.
    */
   readonly securityGroup?: ec2.ISecurityGroup;
+
+  /**
+   * Setting this property informs the CDK that the imported function is in the same environment as the stack.
+   * This affects certain behaviours such as, whether this function's permission can be modified.
+   * When not configured, the CDK attempts to auto-determine this. For environment agnostic stacks, i.e., stacks
+   * where the account is not specified with the `env` property, this is determined to be false.
+   *
+   * Set this to property *ONLY IF* the imported function is in the same account as the stack
+   * it's imported in.
+   * @default - true, if the Stack is configured with an `env` and the account is the same as this function. false, otherwise.
```

Comment:
```suggestion
   * @default - depends: true if the Stack is configured with an explicit `env` (account and region), and the account is the same as this function. For environment-agnostic stacks this will default to `false`, indicating that permissions cannot be added to this function.
```